cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java

```diff
@@ -393,4 +670,137 @@ private String getClassPath(JBallerinaBackend jBallerinaBackend, Package current
+                    HYPHEN + classMetaData[2];
+
+            if (mainJarVsClassMapping.containsKey(mainJarName)) {
+                mainJarVsClassMapping.get(mainJarName).add(className);
+            } else {
+                List<String> classList = new ArrayList<>();
+                classList.add(className);
+                mainJarVsClassMapping.put(mainJarName, classList);
+            }
+        }
+
+        //Modify classes within module jar based on above mapping
+        for (Map.Entry<String, List<String>> mainJarVsClassEntry : mainJarVsClassMapping.entrySet()) {
+
+            mainJarName = mainJarVsClassEntry.getKey();
+            modifiedJarName = mainJarName + HYPHEN + MODIFIED + JAR_EXTENSION;
+
+            for (String testExecutionDependency : testExecutionDependencies) {
+                if (testExecutionDependency.contains(mainJarName) && !testExecutionDependency.contains(TESTABLE)) {
+                    mainJarPath = testExecutionDependency;
+                }
+            }
+            //Add module jar path to classloader URLs
+            classLoaderUrlList.add(mainJarPath);
+            classLoader = AccessController.doPrivileged(
+                    (PrivilegedAction<URLClassLoader>) () -> new URLClassLoader(getURLList(classLoaderUrlList).
+                            toArray(new URL[0]), ClassLoader.getSystemClassLoader()));
+
+            //Modify classes within jar
+            Map<String, byte[]> modifiedClassDef = new HashMap<>();
+            for (String className : mainJarVsClassEntry.getValue()) {
+                List<String> functionNamesList = classVsMockFunctionsMap.get(className);
+                byte[] classFile = getModifiedClassBytes(className, functionNamesList, testSuite, classLoader);
+                modifiedClassDef.put(className, classFile);
+            }
+
+            //Load all classes within module jar
+            Map<String, byte[]> unmodifiedFiles = loadUnmodifiedFilesWithinJar(mainJarPath);
+            String modifiedJarPath = (target.path().resolve(CACHE_DIR).resolve(testSuite.getOrgName()).resolve
+                    (testSuite.getPackageName()).resolve(testSuite.getVersion()).resolve(JAVA_11_DIR)).toString()
+                    + PATH_SEPARATOR + modifiedJarName;
+            //Dump modified jar
+            dumpJar(modifiedClassDef, unmodifiedFiles, modifiedJarPath);
+
+            testExecutionDependencies.remove(mainJarPath);
+            testExecutionDependencies.add(modifiedJarPath);
+        }
+    }
+
+    //Replace unmodified classes with corresponding modified classes and dump jar
+    private void dumpJar(Map<String, byte[]> modifiedClassDefs, Map<String, byte[]> unmodifiedFiles,
+                         String modifiedJarPath) throws IOException {
+        List<String> duplicatePaths = new ArrayList<>();
+        JarOutputStream jarOutputStream = new JarOutputStream(new FileOutputStream(modifiedJarPath));
```

Comment:
Shall we use try-with-resources. It will automatically close the resource without having to add a `finally` block